hack/update-go-format.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${REPO_ROOT}"/hack/util.sh
+
+util::verify_go_version
```

Comment:
```suggestion
```

It seems that `go` is not used here, so there should be no need to check if it exists

Reply:
I understand that `gofmt` depends on the `go` installation, so it needs to be verified
![image](https://user-images.githubusercontent.com/47143494/215313055-8e9cfd17-889e-4bdd-800d-2cf7c59c688c.png)

Reply:
`gofmt` is a standalone binary file and does not depend on `go`

https://github.com/golang/go/blob/master/src/cmd/gofmt/gofmt.go

Reply:
Thanks for tips